spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -16,47 +16,28 @@
 
 package org.springframework.cloud.gcp.pubsub.integration.outbound;
 
-import com.google.cloud.pubsub.spi.v1.Publisher;
-import com.google.pubsub.v1.PubsubMessage;
-import com.google.pubsub.v1.TopicName;
 import java.io.IOException;
-import org.springframework.cloud.gcp.pubsub.integration.converters.SimpleMessageConverter;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
 import org.springframework.integration.handler.AbstractMessageHandler;
 import org.springframework.messaging.Message;
-import org.springframework.messaging.converter.MessageConversionException;
-import org.springframework.messaging.converter.MessageConverter;
 
 /**
- * Converts from internal Spring message to GCP Pub/Sub message and publishes the Pub/Sub
- * message to a Pub/Sub topic.
+ * Sends messages to Google Cloud Pub/Sub by delegating to {@link PubSubTemplate}.
  *
  * @author João André Martins
  */
 public class PubSubMessageHandler extends AbstractMessageHandler {
 
-  private Publisher publisher;
-  private MessageConverter messageConverter;
+  private PubSubTemplate pubsubTemplate;
+  private String topic;
 
-  public PubSubMessageHandler(String projectId, String topicName) throws IOException {
-    this(projectId, topicName, new SimpleMessageConverter());
-  }
-
-  public PubSubMessageHandler(String projectId, String topicName,
-      MessageConverter messageConverter) throws IOException {
-    publisher = Publisher.defaultBuilder(TopicName.create(
-        projectId, topicName)).build();
-    this.messageConverter = messageConverter;
+  public PubSubMessageHandler(String projectId, String topic) throws IOException {
+    pubsubTemplate = new PubSubTemplate(null, projectId);
+    this.topic = topic;
```

Comment:
@joaoandremartins I think it may prove to be more useful if we have a screenshare one day to go over this. The base to grasp is that the integration outbound/inbound channels should only deal with Message from SI. I think this is where you are having some trouble understanding. In SI land Message and Channels are the key constructs that allows components to form a flow.

The PubsubTemplate is just a client to PubSub with some added spring goodness, that the outbound (not so sure about inbound) happens to use.

So think about the oubound part now. You can have a poller like Artem mentioned that polls records from the DB, those are transformed into Message (the currency of SI) and sent to channels, you maybe perhaps have a `header-enricher` somewhere down your flow that adds a header `TOPIC_NAME` to the message. And this message lands on our outbound. The Outbound channel now needs to convert from Message into whatever the contract PubSubTemplate offers (could as well be PubSubMessage) and it then dispatches it via the send method.

The other way is having something like a MessageListenerContainer (as Artem pointed, not needed actually) but an EventDriven source that receives messages from PubSub and then translates them into Message from SI and forwards to the output channel (which would connect the flow to downstream consumers)